README.md

```diff
@@ -315,3 +316,63 @@ Comp 40 gives you the tools to explore a large number of other topics:
+* Understanding Big O (asymptotic analysis) and why certain choices are better (first step towards becoming an ‘engineer’)
+* Reading long documentations (~8 pages PDF) for homework assignments
+* Writing technical comments and documents (README) that make your code easier to follow
+* RECURSION, recursion, ………, Mark loves RECURSION, hence, so should you
+* Debugging is life and you will become master seg fault solver by learning how to use valgrind
+* Visualizing sorting (literally, you will become a number to be sorted in class)
+* Learning to optimize your code and seeing how your choices of code MATTER!
+
+### Tips:
+* Start the homework early and read the homework document early. Comp 15 is a new challenge for comp 11 students and the leap might feel like a lot (this is coming from a comp 11 TA). At times, the document may seem confusing and there may seem like there’s too many things to consider at once. I know that there were students who started homework a few days before and got them done in comp 11. In comp 15, that is NO LONGER an option. You will get STUCK in automatic testing even if your code works with simple input testing. Moreover, if you are confused, so are other students in your class and 2~3 TAs can’t get to over 20 students at once. Really try to start early so that you can discuss your idea with the TA in peace. You will realize that you think a lot clearer when there is no pressure, so really try to start early.
+* Before coding, try to draw out what you would like to do. After drawing out your idea, go talk to a TA to see if your code makes logical sense. Even complex ideas like recursion will all make sense once you begin drawing them out. Spend Friday understanding the documentation from the homework assignment, spend Saturday drawing out your idea, spend Sunday talking with your TA about the sketch that you have and then code! The TAs prefer to talk about your sketch and idea than debug with you because debugging code is not only tiring, it takes a lot longer for you to finish your homework or project. Save the misery on both parts by trying out this strategy! If you have a good design, then you will avoid the mess of getting into bugs that are difficult to find. Moreover, if you have a good design, then the TA will likely not have to look at your code and fix your bug by noticing an error in your design. Trust me on this, you will save a SIGNIFICANT amount of time! 
+* Become acquainted with data structures. Learn how they work and draw them out with the TAs. Seek the TAs to learn how the complex data structures work instead of seeking them solely for debugging! The TAs for comp15 are an awesome group and all have a very deep understanding of how the data structures work. You will see that even one wrong detail in data structures will lead to segmentation fault quicksand or unexpected behavior. Each code you write really matter, so even if you understand, go check with the TAs to make sure you are crystal clear!
```

Comment:
Talking about "quicksand" here is confusing.  And isn't "Become acquainted with data structures" the whole purpose of Comp 15?